hazelcast/src/main/java/com/hazelcast/query/impl/IndexUtils.java

```diff
@@ -52,6 +54,26 @@ public final class IndexUtils {
     /** Pattern to stripe away "this." prefix. */
     private static final Pattern THIS_PATTERN = Pattern.compile("^this\\.");
 
+    /**
+     * Comparator for deserialized HD index entries.
+     */
+    private static final Comparator<HeapData> DESERIALIZED_OFFHEAP_DATA_COMPARATOR = (left, right) -> {
```

Comment:
we should not keep HD-specific code in OS, move to EE repository.

You could also add comments in all places that have to be kept in sync

Reply:
✔️